File: benchmark/c/cpw/write_a_c_program_to_find_the_parity_of_an_unsigned_integer/write_a_c_program_to_find_the_parity_of_an_unsigned_integer.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int write_a_c_program_to_find_the_parity_of_an_unsigned_integer ( unsigned int n ) ;
int write_a_c_program_to_find_the_parity_of_an_unsigned_integer ( unsigned int n ) {
  bool parity = 0;
  while ( n ) {
    parity = ! parity;
    n = n & ( n - 1 );
  }
  return parity;
}

File: translation_result/else_bms/success/c/cpw/write_a_c_program_to_find_the_parity_of_an_unsigned_integer/gemini/BaseRepair/write_a_c_program_to_find_the_parity_of_an_unsigned_integer.rs

fn write_a_c_program_to_find_the_parity_of_an_unsigned_integer(n: u32) -> bool {
    let mut parity = false;
    let mut n = n;
    while n != 0 {
        parity = !parity;
        n = n & (n - 1);
    }
    parity
}
